qbench/fourier/_components/__init__.py

```diff
@@ -1,128 +1,143 @@
-
-        The corresponding circuit is:
-
-        .. code::
-
-                   ┌───┐
-              q_0: ┤ H ├──■──
-                   └───┘┌─┴─┐
-              q_1: ─────┤ X ├
-                        └───┘
-
-        """
-        return self._module.state_preparation()
-
-    @property
-    def u_dag(self) -> Instruction:
-        r"""Unitary $U^\dagger$ defining Fourier measurement.
-
-        The corresponding circuit is:
-
-        .. code::
-
-                 ┌───┐┌───────────┐┌───┐
-              q: ┤ H ├┤ Phase(-φ) ├┤ H ├
-                 └───┘└───────────┘└───┘
-
-        .. note::
-
-           This instruction is needed because on actual devices we can only measure in Z-basis.
-           The $U^\dagger$ unitary changes basis so that subsequent measurement in Z-basis can
-           be considered as performing desired von Neumann measurement to be discriminated from
-           the Z-basis one.
-        """
-
-        return self._module.u_dag(self.phi)
-
-    @property
-    def v0_dag(self) -> Instruction:
-        """Instruction corresponding to the positive part of Holevo-Helstrom measurement.
-
-        The corresponding circuit is:
-
-        .. code::
-
-                 ┌──────────┐┌────────────────┐
-              q: ┤ Rz(-π/2) ├┤ Ry(-φ/2 - π/2) ├
-                 └──────────┘└────────────────┘
-
-        """
-        return self._module.v0_dag(self.phi)
-
-    @property
-    def v1_dag(self) -> Instruction:
-        """Instruction corresponding to the negative part of Holevo-Helstrom measurement.
-
-        The corresponding circuit is:
-
-        .. code::
-
-                 ┌──────────┐┌────────────────┐┌────────┐
-              q: ┤ Rz(-π/2) ├┤ Ry(-φ/2 - π/2) ├┤ Rx(-π) ├
-                 └──────────┘└────────────────┘└────────┘
-        """
-        return self._module.v1_dag(self.phi)
-
-    @property
-    def v0_v1_direct_sum_dag(self) -> Instruction:
-        r"""Direct sum $V_0^\dagger\oplus V_1^\dagger$ of both parts of Holevo-Helstrom measurement.
-
-        .. note::
-           In usual basis ordering, the unitaries returned by this property would be
-           block-diagonal, with blocks corresponding to positive and negative parts
-           of Holevo-Helstrom measurement.
-
-           However, Qiskit enumerates basis vectors in reverse, so the produced unitaries
-           are not block-diagonal, unless the qubits are swapped.
-           See accompanying tests to see how it's done.
-
-           The following article contains more details on basis vectors ordering used
-           (among others) by Qiskit:
-           https://arxiv.org/abs/1711.02086
-        """
-        return self._module.v0_v1_direct_sum(self.phi)
-
-
-_GATESET_MAPPING = {
-    "lucy": _lucy,
-    "rigetti": _rigetti,
-    "ibmq": _ibmq,
-    None: _generic,
-}
+    return 0.5 + 0.25 * np.abs(1 - np.exp(1j * phi))
+
+# """Module defining components used in Fourier discrimination experiment."""
```

Comment:
If this is all moved to `components.py` we can remove the commented out parts.

Reply:
Removed the commented code. Commit 086ecc3